server/src/core/favorite/resolvers/favorite.resolver.ts

```diff
@@ -120,6 +130,62 @@ export class FavoriteResolver {
     });
   }
 
+  @Mutation(() => Boolean, {
+    nullable: false,
+  })
+  @UseGuards(AbilityGuard)
+  @CheckAbilities(CreateFavoriteAbilityHandler)
+  async updateFavoritesOrder(
+    @Args('data') args: FavoriteMutationForUpdatingOrder,
+    @AuthWorkspace() workspace: Workspace,
+  ): Promise<boolean> {
+    const { favoriteId, toIndex } = args;
+
+    const allFavorites = await this.favoriteService.findMany({
+      where: {
+        workspaceId: workspace.id,
+      },
+      include: {
+        person: true,
+        company: {
+          include: {
+            accountOwner: true,
+          },
+        },
+      },
+    });
+    const currentIndex = allFavorites.findIndex(
+      (favorite) => favorite.id === favoriteId,
+    );
+
+    assert(
+      currentIndex !== -1,
+      'Favorite Item with this Id not found',
+      NotFoundException,
+    );
+    // Remove 'favoriteId' favorite from its current position
+    const FavoriteItemToReorder = allFavorites.splice(currentIndex, 1)[0];
+
+    // Insert 'favoriteId' favorite at the 'toIndex'
+    allFavorites.splice(toIndex, 0, FavoriteItemToReorder);
+    // Delete all old favorites
+    await this.favoriteService.deleteMany({
+      where: { workspaceId: workspace.id },
+    });
+
+    const updatedFavoritesData = allFavorites.map((favorite) => ({
+      workspaceId: workspace.id,
+      ...(favorite?.personId ? { personId: favorite.personId } : {}),
+      ...(favorite?.companyId ? { companyId: favorite.companyId } : {}),
+    }));
+
+    await this.favoriteService.createMany({
+      data: updatedFavoritesData,
+    });
```

Comment:
Not sure `findMany` will always return the correct order without `orderBy`. Instead of removing + re-creating all favorites, could you:
- add an `index` property to the `Favorite` model in `schema.prisma`
- generate a prisma migration for this with `yarn prisma:migrate`
- create a `updateOneFavorite` resolver (instead of `updateFavoritesOrder`), that will use `this.favoriteService.update`
- then on reorder, all changed favorites index can be updated one by one, and the `useGetFavoritesQuery` hook can use `orderBy: { index: SortOrder.Asc }`

This is how we manage ordering in backend (for instance for the `ViewField` model).

Reply:
ok sure, i will do this.

Reply:
i added index in schema.prisma and then migrated using yarn prisma:migrate but it is now showing this error when i add new Favorite.

`` The column `favorites.index` does not exist in the current database.
Error: 
Invalid `prisma.favorite.findFirst()` invocation: ``

Schema:

`` model Favorite {
  id    String @id @default(uuid())
  index Int?   @default(0)

  workspaceId       String?
  /// @TypeGraphQL.omit(input: true, output: false)
  person            Person?          @relation(fields: [personId], references: [id], onDelete: Cascade)
  personId          String?
  /// @TypeGraphQL.omit(input: true, output: false)
  company           Company?         @relation(fields: [companyId], references: [id], onDelete: Cascade)
  companyId         String?
  /// @TypeGraphQL.omit(input: true, output: false)
  workspaceMember   WorkspaceMember? @relation(fields: [workspaceMemberId], references: [id])
  workspaceMemberId String?

  @@map("favorites")
}
 ``

Reply:
@thaisguigon @charlesBochet , can you please help me in this?

Reply:
@thaisguigon  @charlesBochet  , can you please help me with this?